core/activity/src/http_proxy.rs

```diff
@@ -59,12 +59,13 @@ async fn proxy_http_request(
 
     let agreement = get_activity_agreement(&db, &activity_id, Role::Requestor).await?;
 
-    // TODO: take care of headers
-    if let Some(value) = request.headers().get(header::ACCEPT) {
-        log::info!("[Header]: {:?}", value);
-    }
-
-    let call_streaming = move |msg| {
+    let http_to_gsb = HttpToGsbProxy {
+        method: method.to_string(),
+        path,
+        body,
+        headers: request.headers().clone(),
+    };
+    let stream = http_to_gsb.pass(move |msg| {
```

Comment:
This should be hidden in library as well. You try to assume that your library doesn't know what gsb is and you must pass method of transfer from outside.
Of course you can keep this separation inside library, but library consumer rather assumes that it should use gsb.